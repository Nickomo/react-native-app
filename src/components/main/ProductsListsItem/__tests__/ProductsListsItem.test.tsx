import { render } from '@testing-library/react-native';
import ProductsListItem from '..';

const testProductsList = {
  id: 'id123',
  name: 'TEST PRODUCTS LIST',
  description: 'product for test',
}

describe('Product Item', () => {
  test('product item render', () => {
    const { getByText } = render(<ProductsListItem productsList={testProductsList}/>)
    const text = getByText('TEST PRODUCTS LIST');
    expect(text).toBeTruthy();
  });
  test('roductsList snapshot testing', () => {
    const productsList = render(<ProductsListItem productsList={testProductsList}/>).toJSON();
    expect(productsList).toMatchSnapshot();
  });
})