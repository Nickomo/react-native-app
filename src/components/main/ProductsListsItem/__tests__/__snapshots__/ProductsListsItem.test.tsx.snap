// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Item roductsList snapshot testing 1`] = `
<View
  collapsable={false}
  handlerTag={3}
  handlerType="PanGestureHandler"
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  style={
    {
      "overflow": "hidden",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "bottom": 0,
        "flexDirection": "row-reverse",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": [
          {
            "translateX": -10000,
          },
        ],
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "center",
          "width": "33%",
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 3,
              "overflow": "hidden",
            },
            {
              "borderRadius": 3,
            },
            {
              "borderRadius": 0,
              "height": "100%",
              "width": "50%",
            },
            false,
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": false,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "orange",
                "borderColor": undefined,
                "borderRadius": 0,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "black",
                    "fontSize": 24,
                  },
                  undefined,
                  {
                    "fontFamily": "material",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "borderRadius": 3,
              "overflow": "hidden",
            },
            {
              "borderRadius": 3,
            },
            {
              "borderRadius": 0,
              "height": "100%",
              "width": "50%",
            },
            false,
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": false,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "red",
                "borderColor": undefined,
                "borderRadius": 0,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "black",
                    "fontSize": 24,
                  },
                  undefined,
                  {
                    "fontFamily": "material",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
    />
  </View>
  <View
    collapsable={false}
    handlerTag={4}
    handlerType="TapGestureHandler"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    pointerEvents="auto"
    style={
      {
        "transform": [
          {
            "translateX": -0,
          },
        ],
      }
    }
  >
    <View
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#bcbbc1",
            "borderTopWidth": 0.5,
            "flexDirection": "row",
            "padding": 14,
          }
        }
        testID="padView"
      >
        <View
          style={
            [
              {
                "alignItems": "flex-start",
                "flex": 1,
                "justifyContent": "center",
              },
              {
                "display": "flex",
              },
            ]
          }
          theme={
            {
              "colors": {
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": {
                  "android": {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              {
                "backgroundColor": "transparent",
                "color": "#242424",
                "fontSize": 20,
                "fontWeight": "bold",
              }
            }
            testID="listItemTitle"
          >
            TEST PRODUCTS LIST
          </Text>
          <Text
            numberOfLines={2}
            style={
              {
                "backgroundColor": "transparent",
                "color": "gray",
                "fontSize": 15,
              }
            }
            testID="listItemTitle"
          >
            product for test
          </Text>
        </View>
        <View
          style={
            {
              "paddingLeft": 16,
            }
          }
        />
        <View
          style={
            {
              "alignSelf": "center",
              "overflow": "hidden",
            }
          }
        >
          <View>
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#D1D1D6",
                      "fontSize": 16,
                    },
                    {
                      "backgroundColor": "transparent",
                    },
                    {
                      "fontFamily": "ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
